docs/resources/User.md

```diff
@@ -14,6 +14,21 @@ data:image/jpeg;base64,BASE64_ENCODED_JPEG_IMAGE_DATA
 
 Ensure you use the proper header type (`image/jpeg`, `image/png`, `image/gif`) that matches the image data being provided.
 
+## Usernames and Nicknames
+
+Limitations on Discord usernames are few and far between. We do, however, have a few of which developers should be mindful. Discord usernames and nicknames are restricted to a minimum of `2` unicode characters and a maximum of `32`.
+
+We also enforce some basic sane data sanitization, like trimming whitespace characters and certain other bad characters. We also block the following substrings and names:
+
+|Invalid Username Substrings| Invalid Usernames|
+|----| ----|
+|@| discordtag
+|#| everyone
+|:| here
+| \```|
+
+There are other rules and restrictions not shared here for the sake of spam and abuse mitigation, but the majority of users won't encounter them. It's important to be cognizent of error messages returned by Discord when editing or updating usernames and nicknames so that edge cases can be handled appropriately.
```

Comment:
This also appears to be overly meaty, boiling it down to the point or throwing it in the recommended list above may be a better option.